libminifi/src/utils/OsUtils.cpp

```diff
@@ -320,9 +316,11 @@ std::string OsUtils::getMachineArchitecture() {
   return "unknown";
 }
 
-}  // namespace utils
-}  // namespace minifi
-}  // namespace nifi
-}  // namespace apache
-}  // namespace org
+std::string OsUtils::getHostName() {
+  char hostname[1024];
+  hostname[1023] = '\0';
+  gethostname(hostname, 1023);
+  return {hostname};
+}
```

Comment:
I don't know how often that happens, probably not very often, but if `gethostname()` returns an error (non-zero) status, then the return value will be a string containing garbage.  We should handle errors in a better way.

Reply:
Updated in 0a19f946e806c662e17edf7ab4959aa5f66da3c2